SECURITY.md

```diff
@@ -0,0 +1,34 @@
+# Security Implication of the Typeshed Stub Uploader
+
+Security for the stub uploader is of the highest importance. If the stub
+uploader gets compromised, an attacker could upload manipulated stub
+packages to gain full access to developer machines or even production hosts.
+Considering the high trust, quick turnaround, and automated installation of
+stub packages, this could have significant security implications.
+
+## Maintainers
+
+TBD
+
+## Typeshed Data
+
+To ensure that a compromised typeshed repository can't lead to compromised
+stub packages, all typeshed data is verified by the stub uploader before
+building packages. The stub uploader ensures that only stub and metadata
+files are added to the stub packages. This also means that no code from the
+typeshed repository must be executed while building packages, and no
+modules must be imported.
+
+## Dependencies
+
+Another possible attack vector are dependencies of stub packages.
+A compromised dependency can have a similar effect to when a stub package
+gets compromised directly. Therefore, only certain dependencies are
+allowed:
+
+* Dependencies on other stub packages created by typeshed.
+* Dependencies on packages the upstream package depends on – even recursively.
+  Since it's likely that a stub package gets installed alongside the
+  upstream package, this does not introduce an additional security liability.
+* Dependencies that are explicitly allowlisted in the stub uploader. These
+  dependencies are vetted to be from a trusted source.
```

Comment:
It would be nice to expand upon this and actually describe and explain the possible attack in https://github.com/typeshed-internal/stub_uploader/pull/61#discussion_r979327370. This kind of hints at how the attack goes, but IMO does not describe it in enough detail. It is the biggest security issue in stub_uploader I'm aware of, and typeshed maintainers need to be aware of it.

Reply:
I'd like to hold off documenting this here at the moment. I kept this section quite vague for the moment, mostly because I think we should resolve and implement #90 first and can then discuss the measures and reasons in this document.

Reply:
Maybe:

```suggestion
  dependencies are vetted to be from a trusted source.

Having one global allowlist is not ideal, because it allows somewhat obscure
dependencies to be added to popular stub packages whose security matters
the most. This may be changed in the near future.
```